packages/usa-date-picker/src/index.js

```diff
@@ -1785,24 +1816,43 @@ const handleShiftPageDownFromDate = adjustCalendar((date) => addYears(date, 1));
 const handleShiftPageUpFromDate = adjustCalendar((date) => subYears(date, 1));
 
 /**
- * display the calendar for the mouseover date.
+ * Set range date classes without re-rendering the calendar. Called when date button is hovered.
+ * Returns early if the date hovered is disabled or if there is a selected date already.
  *
- * @param {MouseEvent} event The mouseover event
- * @param {HTMLButtonElement} dateEl A date element within the date picker component
+ * @param {HTMLElement} dateEl - Calendar date button within the date picker component.
  */
+
 const handleMouseoverFromDate = (dateEl) => {
   if (dateEl.disabled) return;
 
-  const calendarEl = dateEl.closest(DATE_PICKER_CALENDAR);
+  const hoverDate = parseDateString(dateEl.dataset.value);
+  const { calendarEl, selectedDate, rangeDate } = getDatePickerContext(dateEl);
 
-  const currentCalendarDate = calendarEl.dataset.value;
-  const hoverDate = dateEl.dataset.value;
+  if (selectedDate) return;
```

Comment:
Returning early if there is a selected date. This allows us to match current date-range behaviors.